app/src/main/java/com/example/kotlinmigration/database/dao/PostDao.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.kotlinmigration.database.dao
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.Query
+import com.example.kotlinmigration.database.dto.PostDto
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface PostDao {
+
+    @Insert
+    suspend fun addPost(vararg postDto: PostDto)
+
+    @Query("SELECT * FROM posts_table")
+    fun readAllData(): Flow<List<PostDto>> //Removed "suspend" not sure if i should
```

Comment:
the return type here should be `List<PostDto>` not `Flow<List<PostDto>>`

you should keep `suspend` for all database functions
it allows you to call the function in your ViewModel and on the IO thread in an async way

if the amount of data you're moving isnt huge, you probably wont get bit by this, but its a best practice you should use